src/tls_openssl.c

```diff
@@ -881,11 +902,44 @@ static void initialize_underlying_tech(tls_conn_t *conn, SSL *ssl)
     SSL_ctrl(ssl, SSL_CTRL_MODE, SSL_MODE_AUTO_RETRY, NULL);
 }
 
+static int server_alpn_cb(SSL *ssl,
+                          const unsigned char **out,
+                          unsigned char *outlen,
+                          const unsigned char *in,
+                          unsigned inlen,
+                          void *arg)
+{
+    tls_credentials_t *credentials = arg;
+    if (!credentials->allowed_protocols)
```

Comment:
Shouldn't this be an assert? `tls_set_protocol_priority` always allocates a list. And if the callback is invoked after the credentials are destroyed with `destroy_tls_credentials`, then this is a use after free.